src/NATS.Client.Core/NatsOpts.cs

```diff
@@ -43,7 +43,7 @@ public sealed record NatsOpts
     bool Headers,
     NatsAuthOpts AuthOpts,
     NatsTlsOpts TlsOpts,
-    INatsSerializer Serializer,
+    INatsSerializerRegistry Serializers,
```

Comment:
Would change this to be called `SerializerRegistry` to clue the user in that they need to go look for a `SerializerRegistry` type for the option